src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs

```diff
@@ -0,0 +1,259 @@
+            var scriptDirectory = Path.GetDirectoryName(HookIndexPath);
+            var stepHost = HostContext.CreateService<IDefaultStepHost>();
+            Dictionary<string, string> inputs = new()
+            {
+                ["standardInInput"] = JsonUtility.ToString(input),
+                ["path"] = HookIndexPath,
+                ["shell"] = HostContext.GetDefaultShellForScript(HookIndexPath, Trace, prependPath)
+            };
+
+            var handlerFactory = HostContext.GetService<IHandlerFactory>();
+            var handler = handlerFactory.Create(
+                            context,
+                            null,
+                            stepHost,
+                            new ScriptActionExecutionData(),
+                            inputs,
+                            environment: new Dictionary<string, string>(VarUtil.EnvironmentVariableKeyComparer),
+                            context.Global.Variables,
+                            actionDirectory: scriptDirectory,
+                            localActionContainerSetupSteps: null) as ScriptHandler;
+            handler.PrepareExecution(stage);
+
+            IOUtil.CreateEmptyFile(input.ResponseFile);
+            await handler.RunAsync(stage);
+            if (handler.ExecutionContext.Result == TaskResult.Failed)
+            {
+                throw new Exception($"The hook script at '{HookIndexPath}' running command '{input.Command}' did not execute successfully."); // TODO: better exception
+            }
+            var response = GetResponse<T>(input);
+            return response;
+        }
+
+        private string GenerateResponsePath()
+        {
+            return Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Temp), ResponseFolderName, $"{Guid.NewGuid()}.json");
+        }
+
+        private string GetPrependPath(IExecutionContext context)
+        {
+            return string.Join(Path.PathSeparator.ToString(), context.Global.PrependPath.Reverse<string>()); ;
+        }
+
+        private T GetResponse<T>(HookInput input) where T: HookResponse
+        {
+            T response = null;
+
+            if (!string.IsNullOrEmpty(input.ResponseFile) && File.Exists(input.ResponseFile))
+            {
+                response = IOUtil.LoadObject<T>(input.ResponseFile);
+                IOUtil.DeleteFile(input.ResponseFile);
+                Trace.Info($"Response file for the hook script at '{HookIndexPath}' running command '{input.Command}' successfully processed and deleted.");
+                response?.Validate();
+            }
+            else
+            {
+                Trace.Info($"Response file for the hook script at '{HookIndexPath}' running command '{input.Command}' not found.");
+                if (input.Command == HookCommand.PrepareJob)
+                {
+                    throw new Exception($"Response file is required but not found for the hook script at '{HookIndexPath}' running command '{input.Command}'");
```

Comment:
lets somehow catch all of these exceptions and bubble them up to the user is a super friendly way.

Something like
```
Custom container implementation failed with error: {error}. Please contact your self hosted runner administrator.
```